sstables/compaction_manager.cc

```diff
@@ -663,9 +663,11 @@ void compaction_manager::submit_offstrategy(column_family* cf) {
                         _stats.completed_tasks++;
                     } catch (sstables::compaction_stop_exception& e) {
                         cmlog.info("off-strategy compaction was abruptly stopped, reason: {}", e.what());
+                        return put_task_to_sleep(task).then([] {
+                            return make_ready_future<stop_iteration>(stop_iteration::no);
+                        });
```

Comment:
Maybe we should define stopping offstrategy compaction as giving up on reshape. Otherwise the user has no way to stop reshape.